episodes/vignettes.Rmd

```diff
@@ -5,61 +5,76 @@ exercises: 20
-- How can I use R packages to write my research?
+- I have written the functions in my package, but can I also use them?
+- How can I write a tutorial or paper as part of my package?
 :::
 
 ::: objectives
-- Learn how to write reports in the form of a vignette
+- Learn how to write reports in the form of a vignette, using R Markdown.
 :::
 
+Your package now contains functions, and perhaps data.
+Others can easily install your package, and use the functionality you created.
+But, how do they know where to start?
 
+Vignettes are a great way to create a guidebook or tutorial to your package.
+You have probably already seen vignettes, as you were exploring a new package you were interested in using.
 
-## Vignettes
+## Accessing vignettes
 
-If you are reading this, chances are that you are using R for doing research.
-And if you are a researcher, probably you spend a lot of time writing.
-Did you know that you can write your reports, and even your papers, directly from R?
+You can take a look at the vignettes that are available for a package with the function `browseVignettes`.
+For example, this shows you the available vignettes for the `dplyr` package:
 
-We'll do this with vignettes.
-Vignettes are plain-text files that combine enriched text and code.
-Typically, the enriched text is displayed and the code is hidden and executed.
+```r
+browseVignettes("dplyr")
+```
 
-Think about it for a moment.
-A vignette will take code from your package, execute it, generate the figures for you, and put everything together and in context.
-The resulting document would be a human readable website, pdf file or even Microsoft Word file.
+You can also directly open a vignette, if you know its name:
+
+```r
+vignette("dplyr", package="dplyr")
+```
 
 ::: challenge
-## Why would you want to do that?!
-Writing your reports directly in R Studio may sound like an overkill, and it is certainly a bit harder than writing on an external word processor.
-Can you think of a couple of reasons for making the effort worth it?
+## Explore vignettes
+
+Take some time to explore the vignette(s) of your favorite package(s).
+What things do you notice?
+What properties of vignettes stand out to you?
 
 :::: solution
-There are several reasons that make the use of vignettes a very good idea for writing research.
+There are many things that can stand out about vignettes.
+Some things that may be mentioned:
+
+- Vignettes combine text with code, to give a comprehensive overview of the possibilities of a package.
+- Not all packages have vignettes.
+- Some packages have more than one vignette
+- Vignettes can perform many different functions: they can be tutorials, clarify a specific purpose of a package, or even be a scientific report.
+- Code inside a vignette may be executed: the vignette can also contain the results of this code run.
+::::
+:::
+
+A vignette will take code from your package, execute it, generate the figures for you, and put everything together and in context.
```

Comment:
It's not a new sentence, but on first reading it was not clear to me what "put in context" means. does it mean that the vignette explains certain things from the package? does it put different elements from the package into context?